srml/contract/src/gas.rs

```diff
@@ -245,8 +245,12 @@ pub fn refund_unused_gas<T: Trait>(transactor: &T::AccountId, gas_meter: GasMete
 	// Refund gas left by the price it was bought.
 	let b = <balances::Module<T>>::free_balance(transactor);
 	let refund = <T::Gas as As<T::Balance>>::as_(gas_meter.gas_left) * gas_meter.gas_price;
-	<balances::Module<T>>::set_free_balance(transactor, b + refund);
-	<balances::Module<T>>::increase_total_stake_by(refund);
+
+	// this should be infallible since we just charged for it this block. nonetheless, we play it
+	// safe and only issue the refund if issuance doesn't overflow.
+	if let Ok(_) = <balances::Module<T>>::increase_total_stake_by(refund) {
```

Comment:
Because contracts executed can't increase issuance isn't it ?

Reply:
Practically, yes, this shouldn't be able to overflow because:

1. we decrease the total issuance when we buy gas
2. because gas is only spent, the total issuance is increased by lower amount than it was decreased.
3. there are no* code executed between pt.1 and pt.2 beyond the contract module and it doesn't do any transfers.

(* with subtle caveat that the contract module can call callbacks such as `T::DetermineContractAddress` which theoretically could violate this, but we don't care about this cases and most likely will put a warning somewhere in the docs that this is not recommended since it can violate some assumptions here)

Reply:
contracts module will need moving to the new model.